server/metric/mock_metric_test.go

```diff
@@ -209,3 +214,155 @@ func TestMockMetricServer_ListMetricDescriptors(t *testing.T) {
 		t.Errorf("ListMetricDescriptors(%q) == %q, want %q", in, response, want)
 	}
 }
+
+func TestMockMetricServer_GetMetricDescriptorError(t *testing.T) {
+	in := &monitoring.GetMetricDescriptorRequest{}
+	want := validation.ErrMissingField.Err()
+	missingFields := map[string]struct{}{"Name": {}}
+	response, err := client.GetMetricDescriptor(ctx, in)
+	if err == nil {
+		t.Errorf("GetMetricDescriptor(%q) == %q, expected error %q", in, response, want)
+	}
+
+	if !strings.Contains(err.Error(), want.Error()) {
+		t.Errorf("GetMetricDescriptor(%q) returned error %q, expected error %q",
+			in, err.Error(), want)
+	}
+
+	if valid := validation.ValidateErrDetails(err, missingFields); !valid {
```

Comment:
This test statement looks like it belongs in the validation package.

Reply:
`ValidateErrDetails` is actually a utility function in the validation package used here to assert the `err` contiains all of the required `missingFields`